pkg/client/client.go

```diff
@@ -144,6 +145,11 @@ func (c *client) Delete(ctx context.Context, key string, revision int64) error {
 	return nil
 }
 
+func (c *client) Compact(ctx context.Context, revision int64) (int64, error) {
+	resp, err := c.c.Compact(ctx, revision)
+	return resp.Header.GetRevision(), err
```

Comment:
If err isn't nil would we still want to call resp.Header...?

Reply:
The response should [always be non-nil](https://github.com/k3s-io/kine/blob/a9d2c9cfc272ed8e6c6dd88636511b922c7cd296/pkg/server/compact.go#L12). And `GetRevision` handles `Header` being nil, so long as the response itself is non-nil.

Would something like this be preferable?
```golang
resp, err := c.c.Compact(ctx, revision)
if resp != nil {
    return resp.Header.GetRevision(), err
}
return -1, err
```

Reply:
Yeah, it should always be non-nil, but it doesn't hurt to check I guess. If it is nil, I think 0 is better to return than -1. That's an invalid revision and could cause other problems.

Reply:
Resolved in 074e33a3d3961b21f198d533386550fbeb47b7a9.